fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java

```diff
@@ -292,4 +301,93 @@ public long getTableCreateTime(String dbName, String tblName) throws Exception {
         }
         return 0;
     }
+
+    public List<PartitionInfo> getChangedPartitionInfo(Table table, long mvSnapShotID) {
+        LOG.debug("Get changed partitionInfo start, table:{}, mvLatestSnapShotID {}", table, mvSnapShotID);
+        List<PartitionInfo> result = new ArrayList<>();
+        PaimonTable paimonTable = (PaimonTable) table;
+        Map<String, Long> partitionToSnapshotId = fetchChangedPartitionWithVersion(paimonTable, mvSnapShotID);
+        for (Map.Entry<String, Long> entry : partitionToSnapshotId.entrySet()) {
+            PaimonPartitionInfo partitionInfo = new PaimonPartitionInfo(entry.getKey(), entry.getValue());
+            result.add(partitionInfo);
+            if (entry.getValue() != null) {
+                mvSnapShotID = Math.max(mvSnapShotID, entry.getValue());
+            }
+        }
+        LOG.debug("Get changed partitionInfo:{}", result);
+        return result;
+    }
+
+    private Map<String, Long> fetchChangedPartitionWithVersion(PaimonTable paimonTable, long mvSnapshotId) {
+        Map<String, Long> partitionToSnapshotId = new HashMap<>();
+        FileMonitorTable fileMonitorTable = new FileMonitorTable(paimonTable.getNativeTable());
+        Long latestId = fileMonitorTable.snapshotManager().latestSnapshotId();
+        long latestSnapshotId = latestId == null ? Long.MIN_VALUE : latestId;
+        LOG.debug("Paimon table {} latest snapshotId {}, currentId {}",
+                paimonTable.getName(), latestSnapshotId, mvSnapshotId);
+        if (mvSnapshotId >= latestSnapshotId && latestSnapshotId != Long.MIN_VALUE) {
+            LOG.info("Paimon table {} currentId {} > latest snapshotId {} ",
+                    paimonTable.getName(), mvSnapshotId, latestSnapshotId);
+            return partitionToSnapshotId;
+        }
+        ReadBuilder readBuilder = fileMonitorTable.newReadBuilder();
+        StreamTableScan scan = readBuilder.newStreamScan();
+        TableRead read = readBuilder.newRead();
+        if (mvSnapshotId != Long.MIN_VALUE) {
+            scan.restore(mvSnapshotId + 1);
+        } else {
+            scan.restore(null);
+        }
+        try {
+            while (true) {
```

Comment:
Why check `scanMonitorTable` in a loop?  Can it loop too many times?  Be better to add some comments or add a limit to prevent dead-loop or many loops.

Reply:
Because paimon's does not provide an interface to get entire partition info with modify time in once time. MonitorTable only store changed partitions info of once snapshot as one row. So we need to scan all rows in MonitorTable between last mv refresh time to this time by snapshot version.
It may cost too many time to scan if paimon snapshot too frequently.